src/Framework/MockObject/Generator/Generator.php

```diff
@@ -92,11 +92,6 @@ final class Generator
         '__halt_compiler' => true,
     ];
 
-    /**
-     * @psalm-var array<non-empty-string, MockClass>
-     */
-    private static array $cache = [];
-
     /**
```

Comment:
as I have the numbers at hand atm, maybe this is also interessting (even though not related to this PRs topic):

running the test-suite as a whole, these calls are the most memory consuming ones:

![grafik](https://github.com/sebastianbergmann/phpunit/assets/120441/62e20f92-70f4-476d-b640-25a278773024)

and here you can find the slowest calls:

![grafik](https://github.com/sebastianbergmann/phpunit/assets/120441/40d5c5b0-2293-4fff-8e8a-086662bb9f34)

---

recorded with

```
$ php -v
PHP 8.2.12 (cli) (built: Oct 24 2023 21:15:35) (NTS Visual C++ 2019 x64)
Copyright (c) The PHP Group
Zend Engine v4.2.12, Copyright (c) Zend Technologies
    with blackfire v1.92.16~win-x64-non_zts82, https://blackfire.io, by Blackfire
```